src/main.rs

```diff
@@ -14,19 +14,34 @@ use futures::SinkExt;
 mod db;
 mod parser;
 
-#[derive(Debug, Serialize, Deserialize, FromArgs)]
+#[derive(Clone, Debug, Serialize, Deserialize, FromArgs)]
 /// Top level config
 struct RemitsConfig {
-    #[argh(option, short = 'p', default = "\"4242\".to_owned()")]
+    #[argh(option, short = 'p')]
     /// what port to start remits on
-    pub port: String,
+    pub port: Option<String>,
+    #[argh(option, short = 'c')]
+    /// show color in logs
+    pub color: Option<bool>,
 }
 
+impl RemitsConfig {
+    fn update_from_flags(&mut self) {
+        let flags: RemitsConfig = argh::from_env();
+        if flags.port.is_some() {
+            self.port = flags.port;
+        }
+        if flags.color.is_some() {
+            self.color = flags.color;
+        }
+    }
+}
 /// `RemitsConfig` implements `Default`
 impl ::std::default::Default for RemitsConfig {
     fn default() -> Self {
         Self {
-            port: "4242".into(),
+            port: Some("4242".into()),
+            color: Some(true),
```

Comment:
I like the updates, but is this field used for anything right now?

Reply:
Oh i forgot to make note of what i was trying to show. We only use port but i was trying to show how this method of config would scale(poorly) if we have a big config file. I will remove before merging